### PR TITLE
Give verb 

### PR DESCRIPTION
**Pull request by NazarFogmann**
:cl: Moonmandoom
add: Added Give verb
/:cl:

[why]: # because its cool and needed
image

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -307,6 +307,70 @@
...
+/mob/living/carbon/verb/give()
+	set category = "IC"
+	set name = "Give"
+	set src in view(1)
+
+	if(src == usr)
+		to_chat(usr,"<span class='warning'>You feel stupider, suddenly.</span>")
+		return
+
+	if(!ismonkey(src)&&!ishuman(src) || isalien(src) || src.stat || usr.stat || !src.client)
```
Comment (Qustinnus) [inline code/modules/mob/mob.dm]: if something is not a monkey or not a human, then why are you checking seperately if its an alien. the alien would be in the "not a monkey and not a human" category.

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -307,6 +307,70 @@
 	face_atom(A)
 	A.examine(src)
 
+	
+/mob/living/carbon/verb/give()
+	set category = "IC"
+	set name = "Give"
+	set src in view(1)
```
Comment (optimumtact) [inline code/modules/mob/mob.dm]: lol look at this dude

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -307,6 +307,70 @@
...
+
+	if(!usr.canUnEquip(I))
+		return
+
+	var/list/empty_hands = get_empty_held_indexes()
+	if(!empty_hands.len)
+		to_chat(usr,"<span class='warning'>[src]'s hands are full.</span>")
+		return
+
+	switch(alert(src,"[usr] wants to give you \a [I]?",,"Yes","No"))
```
Comment (optimumtact) [inline code/modules/mob/mob.dm]: look at the state of his HAIR

File: code/modules/mob/mob.dm [hunk 4]
```diff
@@ -307,6 +307,70 @@
...
+				to_chat(usr,"<span class='warning'>You need to stay in reaching distance while giving an object.</span>")
+				to_chat(usr,"<span class='warning'>[usr] moved too far away.</span>")
+				return
+
+			if(I != usr.get_active_held_item())
+				to_chat(usr,"<span class='warning'>You need to keep the item in your active hand.</span>")
+				to_chat(usr,"<span class='warning'>[name] seem to have given up on giving [I] to you.</span>")
+				return
+
+			if(src.lying || src.handcuffed)
```
Comment (optimumtact) [inline code/modules/mob/mob.dm]: look at the state of his LIPS

File: code/modules/mob/mob.dm [hunk 5]
```diff
@@ -307,6 +307,70 @@
...
+				to_chat(usr,"<span class='warning'>He is restrained.</span>")
```
Comment (optimumtact) [inline code/modules/mob/mob.dm]: He is restrained

File: code/modules/mob/mob.dm [hunk 6]
```diff
@@ -307,6 +307,70 @@
...
+
+			if(!usr.dropItemToGround(I))
+				return
+
+			if(!put_in_hands(I))
+				to_chat(usr,"<span class='warning'>You can't take [I], so [usr] gave up!</span>")
+				to_chat(usr,"<span class='warning'>[src] can't take [I]!</span>")
+				return
+
+			src.visible_message("<span class='notice'>[usr] handed [I] to [src].</span>")
```
Comment (optimumtact) [inline code/modules/mob/mob.dm]: arbitrary src out the wazoo

**Comment by mattroks101** [comment]
Is this actually copied and pasted from Bay?

File: code/modules/mob/mob.dm (hunk 6 above)
Comment (PKPenguin321) [reply code/modules/mob/mob.dm]: oranges this made me laugh thanks

**Comment by Bawhoppen** [comment]
@MrDoomBringer To be fair, it's slow and clunky to do.

**Comment by MrDoomBringer** [comment]
lmao hahahah this gives them a popup
yeah no

**Comment by Borisvanmemes** [comment]
I was thinking this was a adminverb, it's just the give verb that bay codebases has isn't it?

File: code/modules/mob/mob.dm (hunk 5 above)
Comment (ma44) [reply code/modules/mob/mob.dm]: >assuming genders in [CURRENT_YEAR]

File: code/modules/mob/mob.dm [hunk 7]
```diff
@@ -307,6 +307,70 @@
...
+				return
+
+			if(src.lying || src.handcuffed)
+				to_chat(usr,"<span class='warning'>He is restrained.</span>")
+				return
+
+			empty_hands = get_empty_held_indexes()
+			if(!empty_hands.len)
+				to_chat(usr,"<span class='warning'>Your hands are full.</span>")
+				to_chat(usr,"<span class='warning'>Their hands are full.</span>")
```
Comment (XDTM) [inline code/modules/mob/mob.dm]: You're sending both messages to usr. This goes for all to_chats in the PR.

File: code/modules/mob/mob.dm (hunk 4 above)
Comment (ma44) [reply code/modules/mob/mob.dm]: You could probably hand a person something if they're on the ground unrestrained or forcing it into their hands if stunned/paralyzed

**Comment by Noahnoah55** [comment]
You could make work like catching a thrown object, where if you have the grab intent on it will automatically accept.

**Comment by XDTM** [comment]
yeah but at this point throw intent is pretty much the same

File: code/modules/mob/mob.dm (hunk 3 above)
Comment (ma44) [reply code/modules/mob/mob.dm]: >a alert box with 2 commas

**Comment by ExcessiveUseOfCobblestone** [comment]
how does src not equal usr...

**Comment by Iamgoofball** [comment]
yeah no this is just going to get used to tactical popup spam to distract and stun

**Comment by SouDescolado** [comment]
I think to be viable, remove popup, add an alert ("Someone is trying to give you something. Set to help intent to receive") to chat. So people can only receive if they set to help intent. Additionally, make only be able to give if also on help intent.

**Comment by Iamgoofball** [comment]
Then everyone will walk around with harm to avoid preventing people stuffing illegal items/bombs into their hands

**Comment by ishitbyabullet** [comment]
@Bawhoppen it's perhaps true that giving someone an item is a bit annoying, especially if the hand you put it in suddenly gets an item which then means you have to try giving it again, but we must remember:

If a popup appears then you're distracted or you can't click/move as easily. It may even STOP you from moving. In a server like tgstation where antags can kill for any reason they want without rp, **how long would it take until this was abused?**

Perhaps a nice thing but it will just be abused.

**Comment by PKPenguin321** [comment]
Throw a hyperlink in chat if somebody tries to give you something so they can't tactical popup you

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I have never, ever, ever heard of this being abused nor imagine how would you even do that considering it doesn't block movement or clicks or typing

t. person from server where we've had this for years and is used very commonly

**Comment by Iamgoofball** [comment]
Popups steak window focus and block parts of the screen/some hotkeys, something our powergamers would love

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
We have plenty of lowRP powergamers too and they can't all have failed to notice this.

This is absolutely much ado about nothing but whatever I don't play here, you guys do you

**Comment by Iamgoofball** [comment]
@I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE That's because they'd get banned if they pulled this crap. They wouldn't here.

**Comment by PKPenguin321** [comment]
Chat hyperlink my dudes

**Comment by optimumtact** [comment]
rip and run pr, no response from author, low quality code and an implementation that just won't work here.

**Pull request closed without merging**